### PR TITLE
Added Used-defined APIs template

### PR DESCRIPTION
**Pull request by JorgeFilipeDias**
User-defined APIs template

### CODE REVIEW

File: working/templates/userdefinedAPIsolution/$SCRIPTNAME$_1/$SCRIPTNAME$_1.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+		/// <summary>
+		/// The API trigger.
+		/// </summary>
+		/// <param name="engine">Link with SLAutomation process.</param>
+		/// <param name="requestData">Hold the API request data.</param>
+		[AutomationEntryPoint(AutomationEntryPointType.Types.OnApiTrigger)]
+		public ApiTriggerOutput OnApiTrigger(IEngine engine, ApiTriggerInput requestData)
+		{
+	
+		}
```
Comment (MichielOda) [inline working/templates/userdefinedAPIsolution/$SCRIPTNAME$_1/$SCRIPTNAME$_1.cs:72]:
```suggestion
		{
			// Add custom code here
			
			return new ApiTriggerOutput
			{
				ResponseBody = "Succeeded",
				ResponseCode = (int)StatusCode.Ok
			};
		}
```

Comment (MichielOda) [reply working/templates/userdefinedAPIsolution/$SCRIPTNAME$_1/$SCRIPTNAME$_1.cs:72]: This way it's in line with the snippet that is part of DIS. Maybe double check if the syntax if still fully correct.

**Review by MichielOda — CHANGES_REQUESTED** [review]

File: working/templates/userdefinedAPIsolution/$SCRIPTNAME$_1/$SCRIPTNAME$_1.cs (hunk 1 above)
Comment (JorgeFilipeDias) [reply working/templates/userdefinedAPIsolution/$SCRIPTNAME$_1/$SCRIPTNAME$_1.cs]: Added the snippet from skyline docs

File: .gitignore [hunk 2]
```diff
@@ -1,6 +1,7 @@
 **/bin/*
 **/obj/*
 .vs/*
+/working/.vs/*
```
Comment (MichielOda) [inline .gitignore:4]:
```suggestion
**/.vs/*
```

File: README.md [hunk 3]
```diff
@@ -14,6 +14,10 @@ Template that creates a new connector Visual Studio solution.
 
 Template that creates a new Automation script Visual Studio solution.
 
+### User-defined APIs automation script solution
+
+Template that creates a new User-defined API automation script Visual Studio solution.
```
Comment (MichielOda) [inline README.md:19]:
```suggestion
Template that creates a new User-Defined API Automation script Visual Studio solution.
```

File: working/templates/userdefinedAPIsolution/.template_config/template.json [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  "$schema": "http://json.schemastore.org/template",
+  "author": "Skyline Communications",
+  "classifications": [ "DataMiner" ],
+  "tags": {
+	"language": "C#",
+	"type": "solution",
+	"editorTreatAs":"solution"
+  },
+  "identity": "Skyline.DataMiner.UserDefinedApi.AutomationScript.Solution.Template",
+  "name": "DataMiner User-defined APIs Automation Script Solution",
```
Comment (MichielOda) [inline working/templates/userdefinedAPIsolution/.template_config/template.json:11]:
```suggestion
  "name": "DataMiner User-Defined APIs Automation Script Solution",
```

**Review by MichielOda — CHANGES_REQUESTED** [review]

File: working/templates/userdefinedAPIsolution/$SCRIPTNAME$_1/$SCRIPTNAME$_1.cs [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+	/// <summary>
+	/// Represents a DataMiner Automation script.
+	/// </summary>
+	public class Script
+	{
+		/// <summary>
+		/// The API trigger.
+		/// </summary>
+		/// <param name="engine">Link with SLAutomation process.</param>
+		/// <param name="requestData">Hold the API request data.</param>
```
Comment (Seppe-Vanhee) [inline working/templates/userdefinedAPIsolution/$SCRIPTNAME$_1/$SCRIPTNAME$_1.cs:67]:
```suggestion
		/// <param name="requestData">Holds the API request data.</param>
```

File: .gitignore
Comment (MichielOda) [inline .gitignore:1]: @PedroDebevere Not sure if we actually want this .gitignore? As this contains items specific to Skyline. Also (as this PR will eventually also end up in the main repo (not the Internal one) we want to give the users the choice of what they put in the .gitignore?

**Review by MichielOda — COMMENTED** [review]
Look alright to me for the internal version.
@PedroDebevere, could you have a double look?

File: .gitignore
Comment (PedroDebevere) [reply .gitignore:1]: gitignore can be removed

File: working/templates/userdefinedAPIsolution/.template_config/template.json [hunk 6]
```diff
@@ -0,0 +1,101 @@
+{
+  "$schema": "http://json.schemastore.org/template",
+  "author": "Skyline Communications",
+  "classifications": [ "DataMiner" ],
+  "tags": {
+	"language": "C#",
+	"type": "solution",
+	"editorTreatAs":"solution"
+  },
+  "identity": "Skyline.DataMiner.UserDefinedApi.AutomationScript.Solution.Template",
```
Comment (PedroDebevere) [inline working/templates/userdefinedAPIsolution/.template_config/template.json:10]:
```suggestion
  "identity": "Skyline.DataMiner.UserDefinedApi.Solution.Template",
```

File: working/templates/userdefinedAPIsolution/.template_config/template.json [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+  "name": "DataMiner User-Defined APIs Automation Script Solution",
```
Comment (PedroDebevere) [inline working/templates/userdefinedAPIsolution/.template_config/template.json:11]:
```suggestion
  "name": "DataMiner User-Defined API Solution",
```

File: working/templates/userdefinedAPIsolution/.template_config/template.json [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+  "shortName": "dataminer-user-defined-api-automation-solution",
```
Comment (PedroDebevere) [inline working/templates/userdefinedAPIsolution/.template_config/template.json:12]:
```suggestion
  "shortName": "dataminer-user-defined-api-solution",
```

File: working/templates/userdefinedAPIsolution/.template_config/template.json [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+  "sourceName": "SkylineDataMinerUserDefinedApiAutomationScriptSolutionTemplate",
```
Comment (PedroDebevere) [inline working/templates/userdefinedAPIsolution/.template_config/template.json:13]:
```suggestion
  "sourceName": "SkylineDataMinerUserDefinedApiSolutionTemplate",
```

File: working/templates/userdefinedAPIsolution/.template_config/template.json [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+    "0B4FC455-7B5E-4BB3-A0F4-9841F4D60D6C",
+    "B25F1A08-DDF8-4579-A552-EC81190D8DD7",
+    "7027C535-EB99-4FDF-87C5-43B46A867C2E",
+    "F8938D60-51A0-4880-9CFC-19C66379AE5D",
+    "D8DA8E2C-A4AE-45BE-A120-A18B7078E027",
+    "1AC41DF3-81F2-4A62-8945-DBBB8DC54021",
+    "2BF9AFEF-11BA-4037-827E-A7C4408DD5E0",
+    "54EEBDD9-1D2F-41E5-9654-EB47544FF03D",
+    "37D054BA-8A8F-489C-B707-D45D9F6AF17E",
+	"F2683535-3B81-4454-9E99-120E5016BBCE"
```
[the remark this answers is not in the document]
Comment (JorgeFilipeDias) [reply working/templates/userdefinedAPIsolution/.template_config/template.json:90]: They are used in the .sln file, I've copy pasted from the Automation Script template, should we create new Guids?

File: .gitignore
Comment (JorgeFilipeDias) [reply .gitignore:1]: Removed file

File: working/templates/userdefinedAPIsolution/.template_config/template.json (hunk 10 above)
[the remark this answers is not in the document]
Comment (PedroDebevere) [reply working/templates/userdefinedAPIsolution/.template_config/template.json:90]: Last GUID can be removed, it is not present in the sln file (the GUIDs mentioned in this file denote the GUIDs that need to be replaced by new GUIDs when creating a new solution using this template).

[the remark this answers is not in the document]
Comment (JorgeFilipeDias) [reply working/templates/userdefinedAPIsolution/.template_config/template.json]: Removed the last GUID

**Review by PedroDebevere — APPROVED** [review]

**Pull request merged**